lms/djangoapps/courseware/utils.py

```diff
@@ -95,3 +109,102 @@ def can_show_verified_upgrade(user, enrollment, course=None):
 
     # Show the summary if user enrollment is in which allow user to upsell
     return enrollment.is_active and enrollment.mode in CourseMode.UPSELL_TO_VERIFIED_MODES
+
+
+def _request_financial_assistance(method, url, params=None, data=None):
+    """
+    An internal function containing common functionality among financial assistance utility function to call
+    edx-financial-assistance backend with appropriate method, url, params and data.
+    """
+    financial_assistance_configuration = FinancialAssistanceConfiguration.current()
+    if financial_assistance_configuration.enabled:
+        oauth_application = Application.objects.get(user=financial_assistance_configuration.get_service_user())
+        client = OAuthAPIClient(
+            settings.LMS_ROOT_URL,
+            oauth_application.client_id,
+            oauth_application.client_secret
+        )
+        return client.request(
+            method, f"{financial_assistance_configuration.api_base_url}{url}", params=params, data=data
+        )
+    else:
+        return False, 'Financial Assistance configuration is not enabled'
+
+
+def is_eligible_for_financial_aid(course_id):
+    """
+    Sends a get request to edx-financial-assistance to retrieve financial assistance eligibility criteria for a course.
+
+    Returns either True if course is eligible for financial aid and vice versa.
+    Also returns the reason why the course isn't eligible.
+    In case of a bad request, returns an error message.
+    """
+    response = _request_financial_assistance('GET', f"{settings.IS_ELIGIBLE_FOR_FINANCIAL_ASSISTANCE_URL}{course_id}/")
+    if response.status_code == status.HTTP_200_OK:
+        return response.json().get('is_eligible'), response.json().get('reason')
+    elif response.status_code == status.HTTP_400_BAD_REQUEST:
+        return False, response.json().get('message')
+    else:
+        log.error('%s %s', UNEXPECTED_ERROR_IS_ELIGIBLE, str(response.content))
+        return False, UNEXPECTED_ERROR_IS_ELIGIBLE
+
+
+def get_financial_assistance_application_status(user_id, course_id):
+    """
+    Given the course_id, sends a get request to edx-financial-assistance to retrieve
+    financial assistance application(s) status for the logged-in user.
```

Comment:
logged-in or given user?

Reply:
logged-in. Normally, this method will eventually be called for a user that is logged in.